src/utils/allocators.zig

```diff
@@ -16,13 +18,16 @@ pub fn RecycleFBA(config: struct {
         records: std.ArrayList(Record),
         // for thread safety
         mux: std.Thread.Mutex = .{},
+        // for tracking metrics.
+        maybe_metrics: ?RecycleFBAMetrics = null,
 
         const Record = struct { is_free: bool, buf: [*]u8, len: u64 };
         const AllocatorConfig = struct {
             // used for the records array
             records_allocator: std.mem.Allocator,
             // used for the underlying memory for the allocations
             bytes_allocator: std.mem.Allocator,
+            maybe_metrics: ?RecycleFBAMetrics = null,
```

Comment:
this should take an optional registry which is used to init the metrics (see how the account-cache in accountsdb works)